lib/rules/no-restricted-imports.js

```diff
@@ -86,7 +104,13 @@ const arrayOfStringsOrObjectPatterns = {
                     }
                 },
                 additionalProperties: false,
-                required: ["group"]
+                required: ["group"],
+                not: {
+                    anyOf: [
+                        { required: ["importNames", "allowImportNames"] },
+                        { required: ["importNamePattern", "allowImportNamePattern"] }
```

Comment:
i think we should also validate the pair of
`importNames` and `allowImportNamePattern`
`importNamePattern` and `allowImportNames`

Reply:
Would you add a PR for this please?

The other review suggestions have been applied and committed.

Reply:
have sent the [PR](https://github.com/MichielPater/eslint/pull/3) to your branch.

Reply:
Merged.